subsys/canbus/isotp/Kconfig

```diff
@@ -17,6 +17,24 @@ module = ISOTP
 module-str = ISOTP
 source "subsys/logging/Kconfig.template.log_config"
 
+config ISOTP_USE_CAN_FD
+	bool "Use CAN-FD for ISO-TP"
+	depends on CAN_FD_MODE
+	help
+	  Use CAN-FD for ISO-TP.
```

Comment:
I fought hard to allow for configuring CAN-FD mode at runtime (instead of compile-time) for the CAN controllers (see https://github.com/zephyrproject-rtos/zephyr/issues/45303), so I'm a bit reluctant to add a compile-time option for ISO-TP in this regard. 

Could this be made a run-time configuration with the option to turn it all off via Kconfig (similar as to hiw `CONFIG_CAN_FD_MODE` works)?

Reply:
Yes, that could be done, just need to determine how to best add it to the API.
Currently it would need to be passed to `isotp_bind` and the various `isotp_send` functions

@martinjaeger what are your thoughts on this?

Reply:
@gramsay0 Sorry for the late response.

I would suggest to add a flag to `struct isotp_msg_id` to indicate if FD frames should be used (similar to `struct can_frame` and `struct can_filter`).

The user will have to call `can_set_mode(dev, CAN_MODE_FD);` manually before using ISO-TP. I think it makes sense that the ISO-TP stack does not automatically switch to CAN-FD, as one could use the same CAN device for multiple different purposes. ISO-TP should use the mode already configured for the device.

Reply:
@martinjaeger good idea, the `ide` flag is already in `struct isotp_msg_id` so that makes sense.
I'm on vacation for a couple weeks, feel free to make changes, otherwise I will come back to this later

Reply:
I've added a commit to make CAN-FD configurable at runtime. Please have a look @henrikbrixandersen.